src/game.cpp

```diff
@@ -11983,634 +12019,650 @@ bool game::plmove(int dx, int dy)
-                            mc += (str_req / wheel_indices.size()) *
-                                m.move_cost(grabbed_vehicle->global_x() + grabbed_vehicle->parts[p].precalc[0].x,
-                                        grabbed_vehicle->global_y() + grabbed_vehicle->parts[p].precalc[0].y,
-                                        grabbed_vehicle);
-                        }
-                        //set strength check threshold
-                        //if vehicle has many or only one wheel (shopping cart), it is as if it had four.
-                        if(wheel_indices.size() > 4 || wheel_indices.size() == 1) {
-                            str_req = mc / 4 + 1;
-                        } else {
-                            str_req = mc / wheel_indices.size() + 1;
-                        }
-                    } else {
-                        str_req++;
-                        //if vehicle has no wheels str_req make a noise.
-                        if (str_req <= u.get_str() ) {
-                            sounds::sound( grabbed_vehicle->global_pos3(), str_req * 2,
-                                _("a scraping noise."));
-                        }
-                    }
+    bool diag = trigdist && u.posx() != dest_loc.x && u.posy() != dest_loc.y;
+    const int previous_moves = u.moves;
+    u.moves -= u.run_cost( mcost, diag );
 
-                    //final strength check and outcomes
-                    if (str_req <= u.get_str() ) {
-                        //calculate exertion factor and movement penalty
-                        drag_multiplier += str_req / u.get_str();
-                        int ex = dice(1, 3) - 1 + str_req;
-                        if (ex > u.get_str() ) {
-                            add_msg(m_bad, _("You strain yourself to move the %s!"), grabbed_vehicle->name.c_str() );
-                            u.moves -= 200;
-                            u.mod_pain(1);
-                        } else if (ex == u.get_str() ) {
-                            u.moves -= 200;
-                            add_msg( _("It takes some time to move the %s."), grabbed_vehicle->name.c_str());
-                        }
-                    } else {
-                        u.moves -= 100;
-                        add_msg( m_bad, _("You lack the strength to move the %s"), grabbed_vehicle->name.c_str() );
-                        return false;
-                    }
+    u.burn_move_stamina( previous_moves - u.moves );
 
-                    tileray mdir;
+    // Adjust recoil down
+    u.recoil -= int(u.str_cur / 2) + u.skillLevel( skill_id( "gun" ) );
+    u.recoil = std::max( MIN_RECOIL * 2, u.recoil );
+    u.recoil = int(u.recoil / 2);
+    const int mcost_total = m.move_cost( dest_loc );
+    const int mcost_no_veh = m.move_cost_ter_furn( dest_loc );
+    if( (!u.has_trait("PARKOUR") && mcost_total > 2) || mcost_total > 4 ) {
+        if( veh1 != nullptr && mcost_no_veh != 2 ) {
```

Comment:
This seems to be wrong, it should be `mcost_no_veh == 2`. The special message including the vehicle part name should appear if the terrain (`mcost_no_veh`) does *not* slow down the character. The previous version had `veh1 && m.move_cost(x, y) != 2)`, where `move_cost` includes the vehicle move cost.